### ISSUE TITLE
Add support to customize the KeyStore created by the ClientSslBuilder

### ISSUE THREAD

**Issue by luckyswede** [micronaut-projects/micronaut-core#2941]
I'm setting up a client-side key manager but I dont want the private key to be stored on disk. Instead I have defined my own `ClientSslBuilder` like this:
```
@Singleton
@Replaces(NettyClientSslBuilder.class)
public class XboxClientSslBuilder extends NettyClientSslBuilder {

    @Inject
    public XboxClientSslBuilder(ResourceResolver resourceResolver) {
        super(resourceResolver);
    }

    @Override
    protected KeyStore load(Optional<String> optionalType,
        String resource,
        Optional<String> optionalPassword) throws Exception {
        String type = optionalType.orElse("JKS");
        if (!"custom".equalsIgnoreCase(type)) {
            return super.load(optionalType, resource, optionalPassword);
        }
        KeyStore keyStore = KeyStore.getInstance(KeyStore.getDefaultType());
        // initialize the keystore with cert and private key which I fetch from somewhere else
        return keyStore;
    }
}
```
and I set service properties like this:
```
micronaut.http.services.myservice.ssl.enabled=true
micronaut.http.services.myservice.ssl.client-authentication=NEED
micronaut.http.services.myservice.ssl.key-store.path=notused
micronaut.http.services.myservice.ssl.key-store.type=custom
```

It seems to work well, but is there a better way to do it?
Also, since `NettyClientSslBuilder` is marked as `@Internal` the annotation processor complains while building, but I think thats better than copying the whole class.

**Comment by jameskleeh** [comment]
In the future, please ask questions in [gitter](https://gitter.im/micronautfw) There isn't a better way to do it. If you choose to extend the `NettyClientSslBuilder` then you are opening yourself up to potential breaking changes in minor/patch releases. How are you loading the keystore? Are you using a `LoadStoreParameter`? If so, we can probably allow for something like `micronaut.http.services.myservice.ssl.load-store-parameter: com.example.SomeClass`

**Comment by luckyswede** [comment]
Ok.
My use-case is that I read the client cert from file but the private key from kubernetes secrets, and create the keystore from those.
I have never used `LoadStoreParameter ` so I don't know if that would solve the use case.

**Comment by jameskleeh** [comment]
@luckyswede Can you show the non sensitive parts of the implementation I can see what could be improved

**Comment by luckyswede** [comment]
It basically looks like this
```
        KeyStore keyStore = KeyStore.getInstance(KeyStore.getDefaultType());
        keyStore.load(null);
        X509Certificate cert = SslUtils.cert(CERT_NAME);
        keyStore.setCertificateEntry("cert", cert);
        PrivateKey key = SslUtils.key(KEY_NAME);
        keyStore.setKeyEntry("key", key, null, new Certificate[]{cert});
        return keyStore;
```

**Comment by jameskleeh** [comment]
@luckyswede What about an API like `KeyStoreCustomizer` where you could register a bean that retrieves the keystore and then you could modify it?

**Comment by luckyswede** [comment]
That would solve my case I believe, yes.

**Comment by georgecodes** [comment]
It would be nice if there was a way to just hand the client config a keystore instance you've obtained however you like. I'm trying to configure a truststore from a cert held in a k8s secret but it appears I have to grab it and generate a pkcs12 keystore and point MN at that 

But I do have scenarios - in projects not using MN - where that will literally not be possible. KMS-backed keystores, for instance.

**Comment by dan-turner** [comment]
@luckyswede did you figure out a workaround for this? Encountering the same thing

**Comment by yawkat** [comment]
fixed by #9396 (4.0 release)
